### ISSUE TITLE
`constrained` doesn't work with map schemas

### ISSUE THREAD

**Issue by mullr** [Prismatic/schema#294]
I would expect

```clojure
(s/validate
  (s/constrained 
    {:a s/Int}
    #(odd? (:a %))) 
  {:a 3})
```
to validate, but instead I get:
```clojure
{:type :schema.core/error,
 :schema {:schema {:a
                   {:p? #object[clojure.core$integer_QMARK_ ...],
                    :pred-name integer?}},
          :postcondition #object[clojure.core$odd_QMARK_ ...],
          :post-name odd?},
 :value {:a 3},
 :error (throws? (odd? nil))}
```

This may be illuminating: 
```clojure
(s/validate 
  (s/constrained 
    {:a s/Int} 
    (fn [x] (println x) true)) 
  {:a 3})
```
It appears that the function is always passed `nil` instead of the map `{:a 3}`.

**Comment by w01fe** [comment]
Should be fixed in 1.0.3, thanks for the report!

**Comment by mullr** [comment]
Excellent, thanks!
